test/runtests.jl

```diff
@@ -44,16 +47,19 @@ setprogress!(false)
 
 include(pkgdir(Turing)*"/test/test_utils/AllUtils.jl")
 
+# Collect timing and allocations information to show in a clear way.
+include(pkgdir(Turing)*"/test/test_utils/timing.jl")
+
 @testset "Turing" begin
     @testset "essential" begin
-        include("essential/ad.jl")
+        time_include("essential/ad.jl")
```

Comment:
I'm a bit unhappy about these special commands and additional logic. It took me a while to clean the Turing test suite and remove all custom commands that one had to use for defining and running tests :smile:

Reply:
I get your point. That's a very suitable package for this kind of problem. The reason that I created `time_include` was to avoid duplication in the description and path. Also with TimerOutputs.jl, it will become
```julia
@timeit to "essential/ad.jl" include("essential/ad.jl")
```
which is verbose due to the duplicate "essential/ad.jl". It's slightly less surprising though indeed. Would we have still a less verbose alternative?

> It is also problematic in general to write into the package directory, it might be read-only (IIRC there were also discussions about possibly making them immutable at some point in the future).

It was already in the `.gitignore` so that's why I chose that path. Why is it problematic? What folder would not be problematic? I could do `_performance` since the trailing underscore means "generated" for Franklin and Literate

Reply:
> Why is it problematic?

Because it is not clear if one has write permissions. Running tests is a bit different from generating output for Franklin, Literate, or Documenter: There the intention is clear and the user can be expected to ensure sufficient file system permissions.

> What folder would not be problematic?

E.g. a temporary directory.